apps/arena/native/physics/src/lib.rs

```diff
@@ -100,21 +100,48 @@ fn calculate_triangle_vertices(
 
 #[rustler::nif()]
 fn get_direction_from_positions(position_a: Position, position_b: Position) -> Direction {
-    let x = position_b.x - position_a.x;
-    let y = position_b.y - position_a.y;
-    let len = (x.powi(2) + y.powi(2)).sqrt();
-    Direction {
-        x: x / len,
-        y: y / len,
-    }
+    direction_from_positions(position_a, position_b)
 }
 
 #[rustler::nif()]
 fn calculate_speed(position_a: Position, position_b: Position, duration: u64) -> f32 {
+    let len = distance_between_positions(position_a, position_b);
+    len / duration as f32
+}
+
+#[rustler::nif()]
+fn nearest_entity_direction(entity: Entity, entities: HashMap<u64, Entity>) -> Direction {
+    let mut max_distance = 2000.0;
+    let mut direction = Direction { x: entity.direction.x, y: entity.direction.y };
+
+    for other_entity in entities.values() {
+        if entity.id != other_entity.id {
+            let distance = distance_between_positions(entity.position, other_entity.position);
+            if distance < max_distance {
+                max_distance = distance;
+                direction = direction_from_positions(entity.position, other_entity.position);
```

Comment:
Just a minor comment, with the current implementation your are running `distance_between_positions` for every entity anyways, maybe when you find the first entity in the 2000 distance you can exit the for loop.

Also are we filtering dead players here?